metadata-ingestion/src/datahub/ingestion/source/qlik_sense/data_classes.py

```diff
@@ -0,0 +1,212 @@
+    "description": "",
+    "type": SpaceType.PERSONAL,
+    "createdAt": datetime.now().strftime(QLIK_DATETIME_FORMAT),
+    "updatedAt": datetime.now().strftime(QLIK_DATETIME_FORMAT),
+}
+
+
+class Space(BaseModel):
+    id: str
+    name: str
+    description: str
+    type: SpaceType
+    createdAt: datetime
+    updatedAt: datetime
+    ownerId: Optional[str] = None
+
+    @root_validator(pre=True)
+    def update_values(cls, values: Dict) -> Dict:
+        values[Constant.CREATEDAT] = datetime.strptime(
+            values[Constant.CREATEDAT], QLIK_DATETIME_FORMAT
+        )
+        values[Constant.UPDATEDAT] = datetime.strptime(
+            values[Constant.UPDATEDAT], QLIK_DATETIME_FORMAT
+        )
+        return values
+
+
+class Item(BaseModel):
+    id: str
+    description: str = ""
+    ownerId: str
+    spaceId: str
+    createdAt: datetime
+    updatedAt: datetime
+
+
+class SchemaField(BaseModel):
+    name: str
+    dataType: Optional[str] = None
+    primaryKey: Optional[bool] = None
+    nullable: Optional[bool] = None
+
+    @root_validator(pre=True)
+    def update_values(cls, values: Dict) -> Dict:
+        values[Constant.DATATYPE] = values.get(Constant.DATATYPE, {}).get(Constant.TYPE)
+        return values
+
+
+class QlikDataset(Item):
+    name: str
+    secureQri: str
+    type: str
+    size: int
+    rowCount: int
+    datasetSchema: List[SchemaField]
+
+    @root_validator(pre=True)
+    def update_values(cls, values: Dict) -> Dict:
+        # Update str time to datetime
+        values[Constant.CREATEDAT] = datetime.strptime(
+            values[Constant.CREATEDTIME], QLIK_DATETIME_FORMAT
+        )
+        values[Constant.UPDATEDAT] = datetime.strptime(
+            values[Constant.LASTMODIFIEDTIME], QLIK_DATETIME_FORMAT
+        )
+        if not values.get(Constant.SPACEID):
+            # spaceId none indicates dataset present in personal space
+            values[Constant.SPACEID] = Constant.PERSONAL_SPACE_ID
+        values[Constant.QRI] = values[Constant.SECUREQRI]
+        values[Constant.SIZE] = values[Constant.OPERATIONAL].get(Constant.SIZE, 0)
+        values[Constant.ROWCOUNT] = values[Constant.OPERATIONAL][Constant.ROWCOUNT]
+
+        values[Constant.DATASETSCHEMA] = values[Constant.SCHEMA][Constant.DATAFIELDS]
+        return values
+
+
+class Chart(BaseModel):
+    qId: str
+    qType: str
+
+
+class Sheet(BaseModel):
+    id: str
+    title: str
+    description: str
+    ownerId: str
+    createdAt: datetime
+    updatedAt: datetime
+    charts: List[Chart] = []
+
+    @root_validator(pre=True)
+    def update_values(cls, values: Dict) -> Dict:
+        values[Constant.CREATEDAT] = datetime.strptime(
+            values[Constant.CREATEDDATE], QLIK_DATETIME_FORMAT
+        )
+        values[Constant.UPDATEDAT] = datetime.strptime(
+            values[Constant.MODIFIEDDATE], QLIK_DATETIME_FORMAT
+        )
+        return values
+
+
+class QlikAppDataset(BaseModel):
```

Comment:
do we not get a qri for the dataset?

Reply:
No we don't get QRI for the tables(Mapped with Datahub dataset) which are getting used by Apps